### ISSUE TITLE
Warn if sending unicode text without type=unicode

### ISSUE THREAD

**Issue by timcraft** [Vonage/vonage-ruby-sdk#100]

**Comment by timcraft** [comment]
See also: https://github.com/Nexmo/nexmo-python/issues/42

**Comment by mheap** [comment]
You may find https://github.com/Nexmo/nexmo-php/blob/master/src/Message/EncodingDetector.php useful

**Comment by timcraft** [comment]
@mheap Yep thanks, I've already got GSM7 coded up locally. The sticking point is the behaviour, which would be good to have consistent between client libraries (AFAIK it isn't specced).

What does nexmo-php do? Warn? Raise exception? Auto-set the parameter?

**Comment by mheap** [comment]
You're right, it's not specced

`nexmo-php` makes the user explicitly enable it

```
$msg = new Nexmo\Message\Text($to, $from, $text);
$msg->enableEncodingDetection();
$client->messages()->send($msg);
```

Alternatively, they can use an object that defaults to it enabled

```
$msg = new Nexmo\Message\AutoDetect($to, $from, $text);
$client->messages()->send($msg);
```

Using this method the user has to explicitly opt in to our detection, but if they want to trust us they can use `Nexmo\Message\AutoDetect` rather than setting `type: unicode`

**Comment by timcraft** [comment]
Implemented in df1e9ff (available in v6.0.0)
